core/backend/src/ClassRegistry.ts

```diff
@@ -81,6 +82,26 @@ export class ClassRegistry {
     // the above line creates an anonymous class. For help debugging, set the "constructor.name" property to be the same as the bisClassName.
     Object.defineProperty(generatedClass, "name", { get: () => className });  // this is the (only) way to change that readonly property.
 
+    const navigationProps = Object.entries(entityMetaData.properties)
+      .filter(([_propName, prop]) => prop.isNavigation)
+      .map(([propName, _prop]) => propName);
+
+    if (ClassUtils.isProperSubclassOf(generatedClass, Element)) {
```

Comment:
I don't like this method.  I think it would be better suited to add a 'Is' method to Entity which encapsulates this logic in an EC/BIS way. See the ECClass.is method: https://github.com/iTwin/itwinjs-core/blob/7a6d12bb6e16dc4be407bf8dca9c1d6acf5e4667/core/ecschema-metadata/src/Metadata/Class.ts#L637
It would be like this except implemented against the JS classes instead of the EC Object model

Reply:
sounds reasonable... any reason to not make it a wrapper of this generic utility and keep the utility?
I think the utility's existing placement and tests would be worth keeping

Reply:
also, to keep `ClassUtils` I'd need to also add an `isSubclassOf` method since a class is itself, but a class isn't a proper subclass of itself.